zebra/zebra_dplane.c

```diff
@@ -113,10 +113,15 @@ struct dplane_route_info {
 	struct dplane_nexthop_info nhe;
 
 	/* Nexthops */
+	uint32_t zd_nhg_id;
```

Comment:
I think that 'nexthop_info' set of data is only set up and used for the netlink-oriented nexthop management path. The regular route updates are using the 'route' info struct?

Reply:
the `nexthop_info` struct is included in the `route_info` one.

those access functions are for getting the data out of that.

Reply:
Oh, sorry, I didn't understand your question at first: the existing `get_nhe_id()` api gets a value that a) is only set on netlink platforms, and b) results from a call to `zebra_nhg_resolve()` - so it may not be the id of the actual original `re->nhe` object. I wanted to capture that id.

Reply:
thats true...

okay, fair enough.